packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -536,17 +590,52 @@ class MemoryController extends DisposableController
     return libraryRoot;
   }
 
+  void addAllToNode(
+      Reference root, Map<String, List<HeapGraphClassLive>> allItems) {
+    allItems.forEach((libraryName, classes) {
+      LibraryReference libReference = root.children.singleWhere((library) {
+        return libraryName == library.name;
+      }, orElse: () => null);
+
+      // Library not found add to list of children.
+      libReference ??= LibraryReference(this, libraryName, classes);
+      root.addChild(libReference);
```

Comment:
seems like we are calling addChild again even if the libReference already exists?

Reply:
Yes, I fixed surprised I never saw the problem.